impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.internal.persistence;
+
+import org.ehcache.Cache;
+import org.ehcache.config.persistence.PersistenceConfiguration;
+import org.ehcache.spi.service.LocalPersistenceService;
+import org.ehcache.spi.service.ServiceConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileLock;
+
+/**
+ * @author Alex Snaps
+ */
+public class DefaultLocalPersistenceService implements LocalPersistenceService {
+
+  private final File rootDirectory;
+  private FileLock lock;
+  private File lockFile;
+
+  public DefaultLocalPersistenceService(final PersistenceConfiguration persistenceConfiguration) {
+    rootDirectory = persistenceConfiguration.getRootDirectory();
+  }
+
+  @Override
+  public synchronized void start(final ServiceConfiguration<?> config) {
+    createLocationIfRequiredAndVerify(rootDirectory);
+    try {
+      lockFile = new File(rootDirectory + File.separator + ".lock");
+      final RandomAccessFile rw = new RandomAccessFile(lockFile, "rw");
+      lock = rw.getChannel().lock();
+    } catch (IOException e) {
+      throw new RuntimeException("Couldn't lock rootDir: " + rootDirectory.getAbsolutePath(), e);
+    }
+  }
+
+  @Override
+  public synchronized void stop() {
+    try {
+      lock.release();
+      if (!lockFile.delete()) {
+        // todo log something?;
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Couldn't unlock rootDir: " + rootDirectory.getAbsolutePath(), e);
```

Comment:
Alex and I talked a little on this. I was concentrating on the less important bit there. The exception thrown from release() is certainly more relevant than the outcome of the delete() call.

I can't make up mind if an exception on release() is worthy or re-throw versus log. One part of me says that since it isn't really fatal to stopping that we should just log and continue, the problem will manifest as an exception later iff something starts again in the same directory

deleting the file I guess is nice housekeeping and being a good citizen. failing to delete is certainly not fatal and worthy of warn logging at best imo